py-polars/src/conversion.rs

```diff
@@ -178,6 +178,21 @@ fn struct_dict<'a>(
     dict.into_py(py)
 }
 
+fn decimal_to_digits(v: i128, buf: &mut [u8]) -> usize {
+    const ZEROS: i128 = 0x3030_3030_3030_3030_3030_3030_3030_3030;
```

Comment:
Can you add some comments explaining what this code does and why the `unsafe` is sound?

Reply:
Actually you're right, `buf` should be `&mut [u8; 48]` here, fixed (then unsafe is obviously sound; we could use something like `bytemuck` here I guess to hide unsafe but it's sort of the same thing).

I'll add a few basic tests for this on the Python side as well.